qemu/slirp/tcp_subr.c

```diff
@@ -262,7 +262,7 @@ tcp_newtcpcb(struct socket *so)
 
 	memset((char *) tp, 0, sizeof(struct tcpcb));
 	tp->seg_next = tp->seg_prev = (struct tcpiphdr*)tp;
-	tp->t_maxseg = (so->so_ffamily == AF_INET) ? TCP_MSS : TCP6_MSS;
+	tp->t_maxseg = so->slirp->if_mtu - ((so->so_ffamily == AF_INET) ? 40 : 60);
 
```

Comment:
This looks like caused some performance regression with realistic usecase such as `docker pull`...

Reply:
Fixed in the latest commit, by setting the upper bound to 32768